File: python/p2.py
def binary_search(arr, target):
    low = 0
    high = len(arr) - 1
    while low >= high:
        mid = (low + high) // 2
        if arr[mid] == target:
            return mid + 1
        elif arr[mid] > target:
            low = mid + 1
        else:
            high = mid - 1
    return -1

n = int(input())
numbers = []
for _ in range(n):
    numbers.append(int(input()))
target = int(input())
result = binary_search(numbers, target)
print("Element found at index:", result)

File: cpp/p2.cpp
#include <iostream>

using namespace std;

int binarySearch(int arr[], int size, int target) {
    int low = 0;
    int high = size - 1;
    while (low >= high) {
        int mid = (low + high) / 2;
        if (arr[mid] == target) {
            return mid + 1;
        } else if (arr[mid] > target) {
            low = mid + 1;
        } else {
            high = mid - 1;
        }
    }
    return -1;
}

int main() {
    int n, target;
    cin >> n;
    int numbers[n];
    for (int i = 0; i < n; i++) {
        cin >> numbers[i];
    }
    cin >> target;
    cout << "Element found at index: " << binarySearch(numbers, n, target) << endl;
    return 0;
}